src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+
+public class BlockPaperwall extends Block
+{
+    public static final PropertyBool NORTH = PropertyBool.create("north");
+    public static final PropertyBool EAST = PropertyBool.create("east");
+    public static final PropertyBool SOUTH = PropertyBool.create("south");
+    public static final PropertyBool WEST = PropertyBool.create("west");
+    private static final AxisAlignedBB[] AABB_BY_INDEX = new AxisAlignedBB[]{
+            new AxisAlignedBB(0.4375D, 0.0D, 0.4375D, 0.5625D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.4375D, 0.5625D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.4375D, 0.5625D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.4375D, 0.5625D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 0.5625D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 0.5625D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 0.5625D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 0.5625D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.4375D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.4375D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.4375D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.4375D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D)};
+
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 3F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockPaperwall";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = 1F;
+
+    BlockPaperwall()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.register(this);
+        GameRegistry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    public void addCollisionBoxToList(@NotNull IBlockState Tempstate,
+                                      @NotNull final World worldIn,
+                                      @NotNull final BlockPos pos,
+                                      @NotNull final AxisAlignedBB entityBox,
+                                      @NotNull final List<AxisAlignedBB> collidingBoxes,
+                                      @Nullable final Entity entityIn)
+    {
+        Tempstate = this.getActualState(Tempstate, worldIn, pos);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "Tempstate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1226)